### PR TITLE
5094: URI Format for Ethereum Network Switching

### PR DESCRIPTION
**Pull request by lucemans**
This standard includes all needed information for adding a network to a wallet via URL, by including parameters such as `chainId`, `rpc_url`, `chain_name` and others, such that the network configuration is provided through the URL itself.

As an extension to [EIP-831](./eip-831.md) and neighboring [EIP-681](./eip-681.md) and [EIP-2400](./eip-2400.md), this document aims to standardize the addition of new networks and switching thereof through the means of URLs. User convenience in this case is primary.

### CODE REVIEW

**Comment by lucemans** [comment]
Preview visible at [https://eip5094.luc.page/](https://eip5094.luc.page/)

File: EIPS/eip-5094.md
Comment (MicahZoltu) [inline EIPS/eip-5094.md:102]:
```suggestion
```
This is not a valid section.  Please inline references to relevant standards, and don't link externally.

File: EIPS/eip-5094.md
Comment (MicahZoltu) [inline EIPS/eip-5094.md:15]:
```suggestion
```
Simple Summary is no longer a valid section. See https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md for current template.

File: EIPS/eip-5094.md
Comment (MicahZoltu) [inline EIPS/eip-5094.md:3]:
```suggestion
title: URI Format for Ethereum Network Switching
description: A way of representing various network configurations as URLs.
```
Missing description header field (new, see template for details).

File: EIPS/eip-5094.md
Comment (MicahZoltu) [inline EIPS/eip-5094.md:4]:
```suggestion
author: Luc van Kampen (@lucemans), Jakob Helgesson (@svemat01), Joshua Hendrix (@thejoshuahendrix)
discussions-to: <link to thread on Ethereum Magicians forum>
```
Missing required header field.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]

**Comment by lucemans** [comment]
All changes have been made, currently awaiting post approval for Ethereum Magicians

**Comment by lucemans** [comment]
@MicahZoltu I have made the appropriate changes and updated the Ethereum  Magicians link! Let me know what's next!

**Comment by MicahZoltu** [comment]
Next step is to wait for an ERC editor to review it.  My initial feedback was just a cursory glance to help speed things along, but not a full review.

File: EIPS/eip-5094.md
Comment (SamWilsn) [inline EIPS/eip-5094.md:16]: The title uses `URI`, while the body uses `URL`. I'd suggest picking one for consistency.

File: EIPS/eip-5094.md
Comment (SamWilsn) [inline EIPS/eip-5094.md:22]: Weird sentence structure:

```suggestion
This is understandable, as the `chain_id` provided makes up only one part of what is required to connect to a network. This EIP aims to introduce a new type of URL for usage with deep-linking, QR, and more, to allow users to seamlessly add new networks to their (for ex. mobile) wallet to then be able to more easily partake in `pay-`, `tx-`, or other Ethereum URL interactions.
```

File: EIPS/eip-5094.md
Comment (SamWilsn) [inline EIPS/eip-5094.md:26]:
```suggestion
Introduction of this EIP is meant to bridge to a safer RPC listing system to be introduced in the near future.
```

File: EIPS/eip-5094.md
Comment (SamWilsn) [inline EIPS/eip-5094.md:35]: I'd recommend splitting this into two parts so that it's more obvious that `add` is part of the ERC-831 `payload` (and not `prefix`.)

**Comment by lucemans** [comment]
@SamWilsn I can't seem to be able to re-request review....
But just wanted to say that the changes have been made.

**Review by MicahZoltu — DISMISSED** [review]

**Review by lightclient — APPROVED** [review]

**Review by Villano18 — COMMENTED** [review]
https://www.erc4337.io/docs

**Pull request merged**
